spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/ratelimit/RedisRateLimiter.java

```diff
@@ -138,10 +140,11 @@ public void setApplicationContext(ApplicationContext context) throws BeansExcept
 			throw new IllegalStateException("RedisRateLimiter is not initialized");
 		}
 
-		Config routeConfig = getConfig().getOrDefault(routeId, defaultConfig);
+		Config routeConfig = Optional.ofNullable(getConfig().getOrDefault(routeId, defaultConfig))
```

Comment:
I'm not a fan of using `Optional` for when a simple if else will do.

`Optional` is for use in API return type.

Reply:
done.
Btw I'm a fan of using Optional :) Because I don't like too much `if` & `else`.
e.g. 
```java
    private Optional<String> getTreasureOptionalMap(Exchange exchange) {
        return Optional.ofNullable(exchange)
                .map(Exchange::getSession)
                .map(Session::getRequest)
                .map(Request::getTreasure);
    }
```
replace
```java
    private String getTreasureIf(Exchange exchange) {
        if (null != exchange) {
            Session session = exchange.getSession();
            if (null != session) {
                Request request = session.getRequest();
                if (null != request) {
                    return request.getTreasure();
                } else {
                    return null;
                }
            } else {
                return null;
            }
        } else {
            return null;
        }
    }
```

Reply:
`Optional` is fine in return types and parameters. Not in the middle of a method.